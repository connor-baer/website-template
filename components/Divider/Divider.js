import { css } from '@emotion/core';
import styled from '@emotion/styled';

const baseStyles = ({ theme }) => css`
  border: 3px solid ${theme.colors.p500};
  margin: 2rem 0;
  grid-column: 1 / 13 !important;
  opacity: 0.2;
`;

const Divider = styled('hr')(baseStyles);

export default Divider;
