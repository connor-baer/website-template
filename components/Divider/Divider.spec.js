import React from 'react';

import Divider from './Divider';

describe('Divider', () => {
  function renderDivider(fn, props = {}) {
    return fn(<Divider {...props} />);
  }

  it('should render with default styles', () => {
    const { container } = renderDivider(render);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should meet basic accessibility guidelines', async () => {
    const html = renderDivider(renderToHtml);
    const result = await axe(html);
    expect(result).toHaveNoViolations();
  });
});
