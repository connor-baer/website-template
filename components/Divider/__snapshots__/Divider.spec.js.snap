// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider should render with default styles 1`] = `
.circuit-0 {
  border: 3px solid #f06595;
  margin: 2rem 0;
  grid-column: 1 / 13 !important;
  opacity: 0.2;
}

<hr
  class="circuit-0 circuit-1"
/>
`;
